src/app/containers/Ad/Amp/index.jsx

```diff
@@ -85,50 +86,99 @@ export const AMP_ACCESS_FETCH = service => {
   );
 };
 
+const AdContent = props => {
+  // eslint-disable-next-line react/prop-types
+  const { script, service, dir, label, slotType, pageType } = props;
+  return (
+    <>
+      <StyledLink
+        href={LABEL_LINK}
+        script={script}
+        service={service}
+        dir={dir}
+        tabIndex="-1"
+      >
+        {label}
+      </StyledLink>
+      <AdSlot service={service} slotType={slotType} pageType={pageType} />
+    </>
+  );
+};
+
+const AdWithoutPlaceholder = props => {
+  // eslint-disable-next-line react/prop-types
+  const { ariaLabel, slotType } = props;
+  return (
+    <DisplayWrapper amp-access="toggles.ads.enabled" amp-access-hide="true">
+      <AdSection
+        aria-label={ariaLabel}
+        role="region"
+        data-e2e="advertisement"
+        aria-hidden="true"
+      >
+        <AdContainer slotType={slotType}>
+          <StyledWrapper>
+            <AdContent {...props} />
+          </StyledWrapper>
+        </AdContainer>
+      </AdSection>
+    </DisplayWrapper>
+  );
+};
+
+const AdWithPlaceholder = props => {
+  // eslint-disable-next-line react/prop-types
+  const { ariaLabel, slotType } = props;
+  return (
+    <AdSection
+      aria-label={ariaLabel}
+      role="region"
+      data-e2e="advertisement"
+      aria-hidden="true"
+    >
+      <AdContainer slotType={slotType}>
+        <StyledWrapper>
+          <DisplayWrapper
+            amp-access="toggles.ads.enabled"
+            amp-access-hide="true"
+          >
```

Comment:
Given these two components are identical aside from the introduction of the `DisplayWrapper` could have a shared `Ad` component with a render prop that would contain:
- Without Placeholder
```
            <AdContent {...props} />
```
- With Placeholder:
```
 <DisplayWrapper
            amp-access="toggles.ads.enabled"
            amp-access-hide="true"
          >
            <AdContent {...props} />
          </DisplayWrapper>
```

Just gets rid of some repeated logic and makes it less likely any change made to AdWithoutPlaceholder would not be applied to `AdWithPlaceholder`.

Reply:
I had considered having a shared `Ad` component, however it would mean rendering the `DisplayWrapper` twice when not showing a placeholder...